### PR TITLE
Use libcrypto from OpenSSL

### PR DESCRIPTION
**Pull request by ldechar**
This allow to use Pix4D's openssl package instead of system libcrypto.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,7 @@ include(CheckIncludeFiles)
 include(CheckTypeSize)
 include(GNUInstallDirs)
 include(CheckPIESupported OPTIONAL RESULT_VARIABLE CHECK_PIE_SUPPORTED)
+include(FindOpenSSL)
```
Comment (juanchopanza) [inline CMakeLists.txt:23]: Shouldn't this be something like `find_package(OpenSSL REQUIRED)`?

Comment (ldechar) [reply CMakeLists.txt]: There is no difference of behaviour between `include(FindOpenSSL)` and `find_package(OpenSSL REQUIRED)` that I could see.
However, we are more used to `find_package(OpenSSL REQUIRED)` in Pix4D, so I moved to this one.

File: CMakeLists.txt [hunk 2]
```diff
@@ -199,7 +205,6 @@ if(MSVC)
 else()
 	include(FindPkgConfig)
 	pkg_search_module(CBOR libcbor)
-	pkg_search_module(CRYPTO libcrypto)
 	pkg_search_module(ZLIB zlib)
```
[the remark this answers is not in the document]
Comment (ldechar) [reply CMakeLists.txt:209]: No, the conan packages are picked, you can see this in the conan package build logs:
```
-- CBOR_INCLUDE_DIRS: /home/ldecharr/.conan/data/libcbor/0.9.0-1/pix4d/stable/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include
-- CBOR_LIBRARY_DIRS: /home/ldecharr/.conan/data/libcbor/0.9.0-1/pix4d/stable/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib
-- ZLIB_INCLUDE_DIRS: /home/ldecharr/.conan/data/zlib/1.2.12-1/pix4d/stable/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include
-- ZLIB_LIBRARY_DIRS: /home/ldecharr/.conan/data/zlib/1.2.12-1/pix4d/stable/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib
```

For now I think is not enough value to diverge from original `CMakeLists.txt` to make a change to the `find_package()` yet.

**Review by juanchopanza — APPROVED** [review]
LGTM but I don't fully understand the details of this package.

**Review by odormond — APPROVED** [review]

**Pull request merged**